WooCommerce/Classes/ViewModels/Order Details/Shipping Labels/ShippingLabelOrderItemsAggregator.swift

```diff
@@ -0,0 +1,153 @@
+        guard let productID = productInfo.id else {
+            return .productName(name: productInfo.name)
+        }
+
+        if let product = lookUpProduct(by: productID, products: products) {
+            let orderItem = orderItems.first(where: { $0.productID == productID })
+            return .product(product: product, orderItem: orderItem, name: productInfo.name)
+        } else if let productVariation = lookUpProductVariation(by: productID, productVariations: productVariations) {
+            let orderItem = orderItems.first(where: { $0.variationID == productID })
+            return .productVariation(productVariation: productVariation, orderItem: orderItem, name: productInfo.name)
+        } else {
+            return .productName(name: productInfo.name)
+        }
+    }
+
+    func orderItem(from model: OrderItemModel, quantity: Int, orderItems: [OrderItem]) -> AggregateOrderItem {
+        switch model {
+        case .productName(let name):
+            return .init(productID: 0, variationID: 0, name: name, price: 0, quantity: 0, sku: nil, total: 0, attributes: [])
+        case .product(let product, let orderItem, let name):
+            let productName = orderItem?.name ?? name
+            let price = orderItem?.price ??
+                currencyFormatter.convertToDecimal(from: product.price) ?? 0
+            let totalPrice = price.multiplying(by: .init(decimal: Decimal(quantity)))
+            return .init(productID: product.productID,
+                         variationID: 0,
+                         name: productName,
+                         price: price,
+                         quantity: Decimal(quantity),
+                         sku: orderItem?.sku ?? product.sku,
+                         total: totalPrice,
+                         imageURL: URL(string: product.images.first?.src ?? ""),
+                         attributes: orderItem?.attributes ?? [])
+        case .productVariation(let variation, let orderItem, let name):
+            let productName = orderItem?.name ?? name
+            let price = orderItem?.price ??
+                currencyFormatter.convertToDecimal(from: variation.price) ?? 0
+            let totalPrice = price.multiplying(by: .init(decimal: Decimal(quantity)))
+            return .init(productID: variation.productID,
+                         variationID: variation.productVariationID,
+                         name: productName,
+                         price: price,
+                         quantity: Decimal(quantity),
+                         sku: orderItem?.sku ?? variation.sku,
+                         total: totalPrice,
+                         imageURL: URL(string: variation.image?.src ?? ""),
+                         attributes: orderItem?.attributes ?? [])
+        }
+    }
+
+    func lookUpProduct(by productID: Int64, products: [Product]) -> Product? {
+        products.filter({ $0.productID == productID }).first
```

Comment:
ℹ️ There is also `first()`. 🙂 

```suggestion
        products.first(where: { $0.productID == productID })
```

This makes one question whether we need the function. 🤪

Reply:
hah that's true, I'll still keep them in a function for now. updated in https://github.com/woocommerce/woocommerce-ios/pull/3197/commits/7dbe1f3ebf33f3e571285748bc76007225b1618f